### ISSUE TITLE
How to specify hierarchical filters

### ISSUE THREAD

**Issue by amyisard** [cpietsch/vikus-viewer#41]
Hi,

I see from some previous Issues that it's possible to have hierarchical filters for the keywords, but I haven't found information on how to specify them - is it possible using the main branch code?

Thanks,

Amy

**Comment by cpietsch** [comment]
Hey, I am merging currently some features into main. Did you resolve the filters yourself ?

**Comment by amyisard** [comment]
Hi, I think I originally meant something different by hierarchical filters from what I had seen in the previous issues - we were thinknig of having for example a filter "langage" and when you click on that you see filters "german, english, hebrew, spanish" but we then realised it wasn't clear how that would interact with the other filter behaviour and decided we were happy with the way things were anyway!  And on rereading the other issues I thought they were probably referring to the fact that when you select one keyword, the others only remain visible if they are relevant in the context of the one which has been selected.

**Comment by cpietsch** [comment]
Yes, that should already work. You can try to pull the changes from here: https://github.com/cpietsch/vikus-viewer/compare/master...hierarchical-keywords

Your keywords in data.csv need this format then: level1:level2:level3,level1:level2,....

**Comment by cpietsch** [comment]
If this does not work, let me know

**Comment by amyisard** [comment]
Thanks - I'm going to be away for a couple of weeks but I will try this out when I get back.  Are you planning to merge the hierarchical filters into the main branch?

**Comment by cpietsch** [comment]
I left certain features as branches so that the main Viewer does not get overloaded with options. I wanted to keep some flexibility.  But I see more and more that those features should be integrated modular into the codebase.

**Comment by cpietsch** [comment]
Hi @cweinard, using the branch should work - but maybe something has changed in the config.

I have seen that @sebastianruffberlin did adapt the branch here: https://github.com/sebastianruffberlin/stadtmuseum-viewer (https://sebastianruffberlin.github.io/stadtmuseum-viewer/)

The other route for now would be trying the new crossfilter: https://github.com/cpietsch/vikus-viewer/issues/45

**Comment by cweinard** [comment]
Very cool! I'll give it a try. Thanks!
